### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by wahabwajahat**

### CODE REVIEW

File: angular-client/src/app/components/participants/edit-participant/edit-participant.component.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ export class EditParticipantComponent implements OnInit {
   @Output() cancel = new EventEmitter();
   participantForm: FormGroup;
   phoneregex = /^(\+)?(\d){0,2}(-|.|\s|\()?(\d){3}(-|.|\s|\()?(\d){3}(-|.|\s|\()?(\d){4}$/m;
```
Comment (adriannadiaz) [inline angular-client/src/app/components/participants/edit-participant/edit-participant.component.ts]: The phone regex EVERYWHERE  is not working... 

We should be able to have a space between 
) and 8
(514) 848-2424

The regex is now validating only 10 digits in a row
5148482424 and 
This without spaces...
(514)848-2424

Comment (adriannadiaz) [reply angular-client/src/app/components/participants/edit-participant/edit-participant.component.ts]: I mean Participants, Phonelog, Resources , BOTH for (Add, Edit)

File: angular-client/src/app/components/participants/add-participant/add-participant.component.ts [hunk 2]
```diff
@@ -13,10 +13,12 @@ import { FormGroup, FormControl, Validators, ValidatorFn, FormBuilder, Validatio
   styleUrls: ['./add-participant.component.css']
 })
 export class AddParticipantComponent implements OnInit {
-
+  @ViewChild('f') myNgForm;
+  @Output() addedParticipant = new EventEmitter();
   form: FormGroup;
   socialmedia: FormGroup;
   phoneregex = /^(\d){3}(-|\.|\s|\()?(\d){3}(-|\.|\s|\()?(\d){4}$/m;
```
Comment (adriannadiaz) [inline angular-client/src/app/components/participants/add-participant/add-participant.component.ts]: Change here too.

File: angular-client/src/app/components/resources/add-resource/add-resource.component.ts [hunk 3]
```diff
@@ -11,11 +11,15 @@ import { Router } from '@angular/router';
...
   styleUrls: ['./add-resource.component.css']
 })
+
 export class AddResourceComponent implements OnInit {
   
+  @ViewChild('f') myNgForm;
   resourceTypes = [ 'Housing', 'Medical' ];
+  @Output() addedResource = new EventEmitter();
   form: FormGroup;
   phoneregex = /^(\d){3}(-|\.|\s|\()?(\d){3}(-|\.|\s|\()?(\d){4}$/m;
```
Comment (adriannadiaz) [inline angular-client/src/app/components/resources/add-resource/add-resource.component.ts]: LOL Wahab, I'm putting it here so you can check.

File: angular-client/src/app/components/resources/edit-resource/edit-resource.component.ts [hunk 4]
```diff
@@ -15,6 +15,7 @@ export class EditResourceComponent implements OnInit, OnChanges {
   @Output() cancel = new EventEmitter();
   resourceForm: FormGroup;
   phoneregex = /^(\d){3}(-|\.|\s|\()?(\d){3}(-|\.|\s|\()?(\d){4}$/m;
```
Comment (adriannadiaz) [inline angular-client/src/app/components/resources/edit-resource/edit-resource.component.ts]: Also here

File: angular-client/src/app/components/users/register-user/register-user.component.html [hunk 5]
```diff
@@ -20,7 +20,7 @@
 
         <div class="form-group">
           <mat-form-field fxFlex="100">
-            <input name="rEmail" matInput type="email" placeholder="Email" class="form-control" id="rEmail" [(ngModel)]="user.email" #rEmail="ngModel" required>
+            <input name="rEmail" matInput type="email" placeholder="Email" class="form-control" id="rEmail" [(ngModel)]="user.email" #rEmail="ngModel" required pattern="^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$">
```
Comment (adriannadiaz) [inline angular-client/src/app/components/users/register-user/register-user.component.html]: I don't think the pattern should be here, maybe in the TS file ? @Ldhatter  ?

Comment (Ldhatter) [reply angular-client/src/app/components/users/register-user/register-user.component.html]: yes i concur. @wahabwajahat don't put regex in the html. checkout (Validators Docs)[https://angular.io/api/forms/Validators] there's already an email one.

Comment (wahabwajahat) [reply angular-client/src/app/components/users/register-user/register-user.component.html]: I tried the angular validator for email. It doesn't validate anything after '@' that's why i added my own in html as this is the only template driven form that requires email.

File: angular-client/src/app/components/participants/participants.component.ts [hunk 6]
```diff
@@ -24,4 +25,8 @@ export class ParticipantsComponent implements OnInit {
     this.hasTabChanged = !this.hasTabChanged;
   }
 
+  viewParticipantsTab() {
```
Comment (adriannadiaz) [inline angular-client/src/app/components/participants/participants.component.ts]: Anna fixed this in her PR

File: angular-client/src/app/components/participants/add-participant/add-participant.component.ts [hunk 7]
```diff
@@ -13,10 +13,12 @@ import { FormGroup, FormControl, Validators, ValidatorFn, FormBuilder, Validatio
   styleUrls: ['./add-participant.component.css']
 })
 export class AddParticipantComponent implements OnInit {
-
+  @ViewChild('f') myNgForm;
+  @Output() addedParticipant = new EventEmitter();
   form: FormGroup;
   socialmedia: FormGroup;
-  phoneregex = /^(\d){3}(-|\.|\s|\()?(\d){3}(-|\.|\s|\()?(\d){4}$/m;
+  phoneregex = /^\(?\d{3}\)?[- ]?\d{3}[- ]?\d{4}$/
```
Comment (Ldhatter) [inline angular-client/src/app/components/participants/add-participant/add-participant.component.ts]: does this cover most styles for north american numbers?
+1(514)1234567
1(514)1234567
15141234567
1-514-123-4567
1 514 123 4567
1(514) 123-4567

with or without the 1
with or without ( )
with or without spaces
no letters

Comment (adriannadiaz) [reply angular-client/src/app/components/participants/add-participant/add-participant.component.ts]: you read my mind 💜

**Comment by adriannadiaz** [comment]
screen shot 2018-03-31 at 10 55 42 pm

File: angular-client/src/app/components/participants/add-participant/add-participant.component.ts (hunk 7 above)
Comment (annarog) [reply angular-client/src/app/components/participants/add-participant/add-participant.component.ts]: I made one which covers all the above cases, will push in a minute @wahabwajahat

**Comment by Ldhatter** [comment]
@wahabwajahat  you weren't around to finish what your started so anna did the regex fix. and we completed the merge conflicts.

**Pull request merged**